src/App.tsx

```diff
@@ -1,204 +1,99 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import { TodoMain } from './components/TodoMain';
+import { TodoHeader } from './components/TodoHeader';
+import { FilterType } from './types/FilterType';
+import { TodoNotification } from './components/TodoNotification';
+import { TodoFooter } from './components/TodoFooter';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { ErrorType } from './types/ErrorType';
+
+const USER_ID = 12031;
+
+const filterTodos = (sourceTodos: Todo[], filterBy: FilterType) => {
+  let filteredTodos: Todo[];
+
+  switch (filterBy) {
+    case FilterType.Active:
+      filteredTodos = sourceTodos.filter(todo => !todo.completed);
+      break;
+
+    case FilterType.Completed:
+      filteredTodos = sourceTodos.filter(todo => todo.completed);
+      break;
+
+    default:
+      filteredTodos = [...sourceTodos];
+  }
 
-const USER_ID = 0;
+  return filteredTodos;
+};
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [
+    selectedFilter,
+    setSelectedFilter,
+  ] = useState<FilterType>(FilterType.All);
+
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+
+  const [
+    currentError,
+    setCurrentError,
+  ] = useState<ErrorType>(ErrorType.NoError);
+
+  const onError = (error: ErrorType) => {
+    setCurrentError(error);
+    setTimeout(() => setCurrentError(ErrorType.NoError), 3000);
+  };
+
+  useEffect(() => {
+    setCurrentError(ErrorType.NoError);
+    getTodos(USER_ID)
+      .then(setTodosFromServer)
+      .catch(() => (onError(ErrorType.UnableToLoad)));
+  }, []);
+
+  const [isBeingEdited] = useState(false);
+  const [isBeingSaved] = useState(false);
```

Comment:
If you want to have these values as a state - there should be set functions for them to update the state, if you just want to store these values - it is enough to store them as variables